### PR TITLE
Added GC optimization to the ssz list object

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

File: cl/merkle_tree/hasher.go [hunk 1]
```diff
@@ -83,6 +87,14 @@ func (m *merkleHasher) getBuffer(size int) [][32]byte {
 	return m.internalBuffer[:size]
 }
 
+// getBuffer provides buffer of given size.
+func (m *merkleHasher) getBufferForSSZList(size int) [][32]byte {
+	if size > len(m.internalBufferForSSZList) {
+		m.internalBufferForSSZList = make([][32]byte, size*2)
```
Comment (enriavil1) [inline cl/merkle_tree/hasher.go:93]: Why multiply the size by 2 and not set it as size?

File: cl/merkle_tree/hasher.go [hunk 2]
```diff
@@ -15,10 +16,12 @@ const initialBufferSize = 0 // it is whatever
...
 	// internalBuffer is the shared buffer we use for each operation
-	internalBuffer     [][32]byte
-	internalFlatBuffer []byte
+	internalBuffer           [][32]byte
+	internalFlatBuffer       []byte
+	internalBufferForSSZList [][32]byte
 	// mu is the lock to ensure thread safety
-	mu sync.Mutex
+	mu  sync.Mutex
+	mu2 sync.Mutex // lock onto ssz list buffer
```
Comment (enriavil1) [inline cl/merkle_tree/hasher.go:24]: better to call it sszLock? or sszMu

File: cl/merkle_tree/hasher.go (hunk 1 above)
Comment (Giulio2002) [reply cl/merkle_tree/hasher.go:93]: we make sure to allocate some memory if we increase by one.

so if we append on a slice and we don't have  anything pre-allocated what happens is that you make new slice and append 2x. like this we do it ourselves but without cap only len

File: cl/merkle_tree/hasher.go (hunk 2 above)
Comment (Giulio2002) [reply cl/merkle_tree/hasher.go:24]: so mu2 is better because we can then use It in other places

Comment (Giulio2002) [reply cl/merkle_tree/hasher.go:24]: but ye i guess you right

**Pull request merged**